docs/README.md

```diff
@@ -0,0 +1,327 @@
+# Documentation Index
+
+This page describes the overall organization of documentation for the Swift toolchain.
+It is divided into the following sections:
+
+- [Tutorials](#tutorials)
+  gently guide you towards achieving introductory tasks,
+  while assuming minimal background knowledge.
+- [How-To Guides](#how-to-guides)
+  help you complete common tasks in a step-by-step fashion.
+- [Topic Guides](#topic-guides)
+  discuss key subsystems and concepts,
+  at a high level.
+  They also provide background information
+  and talk about design tradeoffs.
+- [Reference Guides](#reference-guides)
+  contain a thorough technical reference for complex topics.
+  They assume some overall understanding of surrounding subsystems.
+- [Recommended Practices](#recommended-practices)
+  suggests guidelines for writing code and diagnostics.
+- [Project Information](#project-information)
+  tracks continuous integration (CI), branching and release history.
+- [Evolution Documents](#evolution-documents)
+  includes proposals and manifestos for changes to Swift.
+- The [External Resources](#external-resources) section
+  provides links to valuable discussions about Swift development,
+  in the form of talks and blog posts.
+- The [Uncategorized](#uncategorized) section is for
+  documentation which does not fit neatly into any of the above categories.
+  We would like minimize things in this section;
+  avoid adding new documentation here.
+
+Sometimes documentation is not enough.
+Especially if you are a new contributor, you might run into roadblocks
+which are not addressed by the existing documentation.
+Or they are addressed somewhere but you cannot find the relevant bits.
+If you are stuck,
+please use the [development category][] on the Swift forums to ask for help!
+
+Lastly, note that
+we are slowly moving towards a more structured form of documentation,
+inspired by the Django project [[1][Django-docs-1]] [[2][Django-docs-2]].
+Hence parts of this page are aspirational
+and do not reflect how much of the existing documentation is written.
+Pull requests to clean up the [Uncategorized](#uncategorized) section,
+or generally fill gaps in the documentation are very welcome.
+If you would like to make major changes,
+such as adding entire new pieces of documentation,
+please create a thread on the Swift forums
+under the [development category][] to discuss your proposed changes.
+
+[development category]: https://forums.swift.org/c/development
+[Django-docs-1]: https://docs.djangoproject.com/
+[Django-docs-2]: https://documentation.divio.com/#the-documentation-system
+
+## Tutorials
```

Comment:
First of all, this is really great work! Good job! 
Secondly, I see that there are no external links to anything outside of the repo is mentioned e.g. articles/ talks/ YouTube videos. Is there a reason why?

IMHO I think talks is much easier to watch for beginners. What do you think?

Reply:
The reason is that it's not yet ready! 😅 I have a prototype up here: https://github.com/varungandhi-apple/swift/blob/vg-doc-wars/docs/ExternalResources.md

It's very much based on what you wrote up earlier; I tried to go through all the talks and add more details. There are still some things which are missing, which is why I don't have a PR up for it. The tentative plan is:

1. Merge the Index. (this PR)
2. Merge the External Resources page.
3. Merge the Getting Started guide (prototype: https://github.com/varungandhi-apple/swift/blob/vg-doc-wars/docs/HowToGuides/GettingStarted.md)
4. Merge documentation describing the compiler pipeline and different modes (prototype: https://github.com/varungandhi-apple/swift/blob/vg-doc-wars/docs/TopicGuides/CompilerArchitecture.md).

Reply:
This is really awesome! I'd love if you can mention me in the upcoming PRs 😄

Side note: I really like it and love your info about `sccache`.

Reply:
Sure thing, I will ping you. 😄